metador_core/widget/__init__.py

```diff
@@ -107,6 +103,18 @@ def __init__(
             else:
                 raise ValueError("The node does not contain '{schema_name}' metadata!")
 
+        # resizing images to fit within set bounds
+        if self._meta.width is not None and self._meta.height is not None:
+            scale_factor = min(
+                max_height / self._meta.height.value, max_width / self._meta.width.value
+            )
+            max_width = int(self._meta.width.value * scale_factor)
+            max_height = int(self._meta.height.value * scale_factor)
+
+        # maximal width and height to use / try to fill
+        self._w = max_width
+        self._h = max_height
+
```

Comment:
Is this scaling specific for image widgets? Then it does not belong into the abstract widget class and should go into the image widget.

Reply:
Scaling applies, in different amounts, not only to images but also some other kinds of widgets. Which is what is handled here. I will update the comment accordingly.